src/map/mob.cpp

```diff
@@ -5810,8 +5807,10 @@ static bool mob_parse_row_mobskilldb(char** str, int columns, int current)
 	else
 		ms->emotion = -1;
 
-	if(str[18] != NULL && mob_chat(atoi(str[18]))!=NULL)
-		ms->msg_id = atoi(str[18]);
+	if (*str[18]) {
+		uint16 id = static_cast<uint16>(atoi(str[18]));
+		ms->msg_id = ((mob_chat_db.find(id) != nullptr) ? id : 0);
```

Comment:
If you already touch this:
replace atoi with strtol and give a proper error message if id is != 0 and not found please.